### PR TITLE
Change issue templates to forms

### PR DESCRIPTION
**Pull request by emdneto**
This PR proposes to transition otel-python issue templates to GitHub's [issue forms feature](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms).

The same was done in the main repo https://github.com/open-telemetry/opentelemetry-python/pull/3951. Consolidating the forms here too

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    id: environment
+    attributes:
+      label: Describe your environment
+      description: |
+        Please describe any aspect of your environment relevant to the problem, including your Python version, [platform](https://docs.python.org/3/library/platform.html), version numbers of installed dependencies, information about your cloud hosting provider, etc. If you're reporting a problem with a specific version of a library in this repo, please check whether the problem has been fixed on main.
+      value: |
+        OS: (e.g, Ubuntu)
+        Python version: (e.g., Python 3.8.10)
+        SDK version: (e.g., 1.25.0)
+        API version: (e.g., 1.25.0)
```
Comment (xrmx) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:23]: I think here is more interesting to ask for an package version: (e.g. 0.46.0)

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    attributes:
+      label: What happened?
+      description: Please provide as much detail as you reasonably can.
+    validations:
+      required: true
+
+  - type: textarea
+    attributes:
+      label: Steps to Reproduce
+      description: Provide a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example). We will close the issue if the repro project you share with us is complex or we cannot reproduce the behavior you are reporting. We cannot investigate custom projects, so don't point us to such, please.
```
Comment (xrmx) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:35]:
```suggestion
      description: Provide a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) if possible and the needed steps to reproduce the problem.
```

File: .github/ISSUE_TEMPLATE/feature_request.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+labels: [feature-request]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Before opening a feature request against this repo, consider whether the feature should/could be implemented in the [other OpenTelemetry client libraries](https://github.com/open-telemetry/). If so, please [open an issue on opentelemetry-specification](https://github.com/open-telemetry/opentelemetry-specification/issues/new) first.
+  - type: textarea
+    id: related-problem
+    attributes:
+      label: Is your feature request related to a problem?
```
Comment (xrmx) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:13]: I would rephrase it to: What problem do you want to solve?

Comment (emdneto) [reply .github/ISSUE_TEMPLATE/feature_request.yaml:13]: Excellent. I liked it. When we have the final review here, I'll fix this in the main repo, too to keep forms similar.

**Review by lzchen — APPROVED** [review]
